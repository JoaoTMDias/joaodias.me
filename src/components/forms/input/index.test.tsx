import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { TextInput, defaultProps } from "./index";

afterEach(cleanup);

describe("<TextInput />", () => {
	it("should render without errors", () => {
		const component = render(<TextInput id="name" value="" {...defaultProps} />);

		expect(component).toMatchSnapshot();
	});

	it("should call onChange when typing", async () => {
		const onChangeMock = jest.fn();
		const { findByTestId } = render(<TextInput {...defaultProps} id="name" value="" onChange={onChangeMock} />);

		const input = await findByTestId("component-text-input-name");

		fireEvent.change(input, {
			target: {
				value: "teste",
			},
		});

		expect(onChangeMock).toHaveBeenCalled();
	});

	it("should have isFocused class when focused", async () => {
		const { findByTestId } = render(<TextInput {...defaultProps} id="name" value="" />);

		const input = await findByTestId("component-text-input-name");
		const wrapper = await findByTestId("component-text-wrapper");

		fireEvent.focus(input);

		expect(wrapper).toHaveClass("isFocused");
	});

	it("should not have isFocused class when loses focus", async () => {
		const { findByTestId } = render(<TextInput {...defaultProps} id="name" value="" />);

		const input = await findByTestId("component-text-input-name");
		const wrapper = await findByTestId("component-text-wrapper");

		fireEvent.blur(input);

		expect(wrapper).not.toHaveClass("isFocused");
	});

	it("should have helper text", async () => {
		const helperText = "This a helper text";
		const { findByTestId } = render(<TextInput {...defaultProps} id="name" value="" helperText={helperText} />);

		const helper = await findByTestId("component-text-helper");

		expect(helper).toHaveTextContent(helperText);
	});
});
