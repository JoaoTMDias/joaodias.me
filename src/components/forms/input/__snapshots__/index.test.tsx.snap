// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm cktUFz"
        data-form="input"
        data-testid="component-text-wrapper"
      >
        <label
          aria-labelledby="-text-input-id"
          class="content"
          data-testid="component-text-label"
          for="text-input-id"
        >
          <span
            class="label"
            id="-text-input-id"
          />
          <input
            class="input undefined"
            data-testid="component-text-input"
            id="text-input-id"
            maxlength="50"
            minlength="3"
            name="text-input-id"
            placeholder="text-input-placeholder"
            type="text"
            value=""
          />
          <p
            class="helper"
            data-testid="component-text-helper"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm cktUFz"
      data-form="input"
      data-testid="component-text-wrapper"
    >
      <label
        aria-labelledby="-text-input-id"
        class="content"
        data-testid="component-text-label"
        for="text-input-id"
      >
        <span
          class="label"
          id="-text-input-id"
        />
        <input
          class="input undefined"
          data-testid="component-text-input"
          id="text-input-id"
          maxlength="50"
          minlength="3"
          name="text-input-id"
          placeholder="text-input-placeholder"
          type="text"
          value=""
        />
        <p
          class="helper"
          data-testid="component-text-helper"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
